meson-scripts/build_libbpf

```diff
@@ -7,18 +7,27 @@ out=${out#\"}
 out=${out%\"}
 args=($out)
 
+declare -a cflags=()
+
 idx=0
 cc=${args[idx]}
 if [ "$cc" = "ccache" ]; then
     idx=$((idx+1))
     cc="$cc ${args[idx]}"
 fi
 
-declare -a cflags=()
+# This fixes an issue that was happening in some cases when building with gcc:
+# relocation R_X86_64_32 against `.rodata' can not be used when making a PIE
+# object; recompile with -fPIE
+if [ "$cc" = "cc" ]; then
+    cflags+="-fPIE "
+fi
```

Comment:
I'm not understanding what the meaning is of `-fPIE` for a static library. My guess is that this implies `-fPIC` or something? Anyways, I think if we're going to do this we should just always build with `-fPIE`. Clearly it's something that rustc requires.

Reply:
Reads to me it's position independent but can't be linked to create dynamic libs. No idea why it's needed.

Reply:
Yeah, I've just never seen `-fPIE` used to create a lib. I thought it was always `-fPIC` given that it's not creating an executable.